server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java

```diff
@@ -200,26 +206,31 @@ private void createQueuedEventsSpans(SRunningBuild build, Span buildSpan) {
                 List<String> keySplitList = Pattern.compile(":")
                         .splitAsStream(key)
                         .collect(Collectors.toList());
-                this.otelHelper.addAttributeToSpan(childSpan, PluginConstants.ATTRIBUTE_NAME, keySplitList.get(1));
-                this.otelHelper.addAttributeToSpan(childSpan, PluginConstants.ATTRIBUTE_SERVICE_NAME, keySplitList.get(0));
+                setQueuedEventsSpanBuildAttributes(build, childSpan, keySplitList.get(1), keySplitList.get(0));
                 childSpan.end(startDateTime + value.longValue(), TimeUnit.MILLISECONDS);
                 Loggers.SERVER.debug("OTEL_PLUGIN: Queued span added");
                 startDateTime+= value.longValue();
             }
         }
     }
 
+    private void setQueuedEventsSpanBuildAttributes(SRunningBuild build, Span span, String spanName, String spanServiceName) {
```

Comment:
This looks very similar to `setSpanBuildAttributes`... Any particular reason for the duplication?

Reply:
I can try to amalgamate this. Give me a bit.